src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIRequestBuilder.java

```diff
@@ -48,9 +49,17 @@ public class ServiceNowTableAPIRequestBuilder extends RestAPIRequest.Builder {
    */
   private static final String METADATA_API_URL_TEMPLATE = "%s/api/now/ui/meta/%s";
 
-  public ServiceNowTableAPIRequestBuilder(String instanceBaseUrl, String tableName, boolean isSchemaRequired) {
+  public ServiceNowTableAPIRequestBuilder(String instanceBaseUrl, String tableName, boolean isSchemaRequired,
+                                          SchemaType schemaType) {
     if (isSchemaRequired) {
-      this.setUrl(String.format(METADATA_API_URL_TEMPLATE, instanceBaseUrl, tableName));
+      if (SchemaType.METADATA_API_BASED == schemaType) {
```

Comment:
Please simplify this 

move `this.setUrl(String.format(TABLE_API_URL_TEMPLATE, instanceBaseUrl, tableName));` before the `if` block  as the default